src/main/java/com/epam/esm/controller/OrderController.java

```diff
@@ -0,0 +1,68 @@
+package com.epam.esm.controller;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+import com.epam.esm.dto.OrderDTO;
+import com.epam.esm.dto.OrderRequestDTO;
+import com.epam.esm.service.OrderService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.hateoas.PagedModel;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/order")
+public class OrderController {
+
+  private final OrderService orderService;
+
+  public OrderController(OrderService orderService) {
+    this.orderService = orderService;
+  }
+
+  @GetMapping
+  public ResponseEntity<PagedModel<EntityModel<OrderDTO>>> getAllOrders(Pageable pageable, PagedResourcesAssembler<OrderDTO> assembler) {
```

Comment:
Similarly in this class you user third approach to building HATEOAS + pageable. Please, unify all controllers yo use same approach.

Reply:
I've unified all methods